python/lib/core/dmod/core/dataset.py

```diff
@@ -335,8 +228,10 @@ def extend_life(self, value: Union[datetime, timedelta]) -> bool:
         if not self.is_temporary:
             return False
         elif isinstance(value, timedelta):
-            self._set_expires(self._expires + value)
+            # TODO: Fix bug. expires could be None
```

Comment:
@robertbartel should `self._expire` have a default value?

Reply:
It should have a default value of `None`, if the dataset is not temporary. I'm not entirely sure whether that's what you were asking, or if you meant a default _datetime_ value.

Reply:
The initial intent was for non-temporary datasets to not have an expire time, as this intuitively makes sense.  We could change how this is represented, though, if there is good reason.

Reply:
I see 232 is unreachable if `self._expires` is `None`. I think I marked this as a TODO on a first quick pass and my bad for not double checking the logic in the `is_temporary`  property. Ill remove these comments in the source.